### PR TITLE
feat(deploy): release & document deploy command

### PR DESCRIPTION
**Pull request by ThibodeauJF**
https://coveord.atlassian.net/browse/KIT-2308

There is also a config file format like the following, how should we document this in the repo?
```typescript
interface FileInput {
  path: string;
}

interface JavaScriptFileInput extends FileInput {
  isModule: boolean;
}

export interface DeployConfig {
  name: string;
  dir: string;
  htmlEntryFile: FileInput;
  javascriptEntryFiles?: JavaScriptFileInput[];
  javascriptUrls?: JavaScriptFileInput[];
  cssEntryFiles?: FileInput[];
  cssUrls?: FileInput[];
}
```

Example (coveo.deploy.json)
```json
{
  "name": "my page",
  "dir": "dist-folder",
  "htmlEntryFile": {
    "path": "index.html"
  },
  "javascriptEntryFiles": [
    {
      "path": "index.js",
      "isModule": false
    }
  ],
  "javascriptUrls": [
    {
      "path": "https://static.cloud.coveo.com/atomic/v2/atomic.esm.js",
      "isModule": true
    }
  ],
  "cssEntryFiles": [
    {
      "path": "css/index.css"
    }
  ],
  "cssUrls": [
    {
      "path": "https://static.cloud.coveo.com/atomic/v2/themes/coveo.css"
    }
  ]
}
```

### CODE REVIEW

**Review by jpmarceau — APPROVED** [review]
Very clear JF 👌

**Comment by jpmarceau** [comment]
About the other config file format, I'm not sure.
It could probably be in a usage article on [the docs site](https://docs.coveo.com/en/cli/), but in the repo itself I'm not sure. Could you create an example file and link to it from the description of your command?

**Comment by ThibodeauJF** [comment]
@jpmarceau Done !

File: packages/cli/core/README.md [hunk 1]
```diff
@@ -1043,6 +1044,34 @@ EXAMPLES
...
+
+FLAGS
+  -p, --pageId=7944ff4a-9943-4999-a3f6-3e81a7f6fb0a  The hosted page ID.
+  -c, --config=./configs/myconfig.json [default: coveo.deploy.json] The path to the configuration file.
+  -o, --organization=myOrgID  The identifier of the organization inside which to perform operations. See
+                              <https://docs.coveo.com/en/1562/#organization-id-and-other-information>.
+
+DESCRIPTION
+  Deploy your search application to the Coveo infrastructure.
+
```
Comment (jpmarceau) [inline packages/cli/core/README.md:1063]: I think I went a bit too quickly during my first review and didn't understand enough about the config; apologies. Now that I think a bit more about it, I wonder if we expect people to craft the config.json themselves in some cases? If we do, I think we should explain what each param does. We would probably have to do it in this description, something like the following. (I'm not sure how to describe the params)

```suggestion
The `config` must contain the following parameters:

{
  "name": "The name of the hosted search page.",
  "dir": "The directory of the hosted search page.",
  "htmlEntryFile": {
    "path": "The HTML file that ... "
  },
  "javascriptEntryFiles": [
    {
      "path": "The path to ...",
      "isModule": "Whether ... "
    }
  ],
  "javascriptUrls": [
    {
      "path": "https://static.cloud.coveo.com/atomic/v2/atomic.esm.js",
      "isModule": true
    }
  ],
  "cssEntryFiles": [
    {
      "path": "css/index.css"
    }
  ],
  "cssUrls": [
    {
      "path": "https://static.cloud.coveo.com/atomic/v2/themes/coveo.css"
    }
  ]
}

```

Would that make sense or would it just be overkill?

Comment (ThibodeauJF) [reply packages/cli/core/README.md:1063]: Sure we can do that! I'll fill it out

File: packages/cli/core/README.md [hunk 2]
```diff
@@ -1043,6 +1044,65 @@ EXAMPLES
...
+## `coveo ui:deploy`
+
+Deploy your search application to the Coveo infrastructure.
+
+```
+USAGE
+  $ coveo ui:deploy [-p <value>] [-c <value>] [-o <value>]
+
+FLAGS
+  -p, --pageId=7944ff4a-9943-4999-a3f6-3e81a7f6fb0a  The hosted page ID.
```
Comment (btaillon) [inline packages/cli/core/README.md:1056]: If we don't specify a page ID, will it create a new page?

File: packages/cli/core/README.md [hunk 3]
```diff
@@ -1043,6 +1044,65 @@ EXAMPLES
...
+  -c, --config=./configs/myconfig.json [default: coveo.deploy.json] The path to the configuration file.
+  -o, --organization=myOrgID  The identifier of the organization inside which to perform operations. See
+                              <https://docs.coveo.com/en/1562/#organization-id-and-other-information>.
```
Comment (btaillon) [inline packages/cli/core/README.md:1059]: If we don't specify an organization, which one will be used?

File: packages/cli/core/README.md [hunk 4]
```diff
@@ -1043,6 +1044,65 @@ EXAMPLES
...
+
+DESCRIPTION
+  Deploy your search application to the Coveo infrastructure.
+Them target `config` if present, must contain the following parameters:
```
Comment (btaillon) [inline packages/cli/core/README.md:1063]: I'm not sure what this is meant to say 🤔 

```suggestion
  Deploy your search application to the Coveo infrastructure.
  The target `config`, if present, must contain the following parameters:
```

**Review by btaillon — APPROVED** [review]

File: packages/cli/core/README.md (hunk 3 above)
Comment (ThibodeauJF) [reply packages/cli/core/README.md:1059]: The one you're logged in w/ the cli

File: packages/cli/core/README.md (hunk 4 above)
Comment (ThibodeauJF) [reply packages/cli/core/README.md:1063]: Yes I need to update this whole file

**Review by y-lakhdar — APPROVED** [review]

File: packages/cli/core/README.md [hunk 5]
```diff
@@ -1043,6 +1044,66 @@ EXAMPLES
...
+  -c, --config=./configs/myconfig.json [default: coveo.deploy.json] The path to the configuration file.
+  -o, --organization=myOrgID  The identifier of the organization inside which to perform operations. If not specified, the CLI logs you in to
+                              the first available organization. See
+                              <https://docs.coveo.com/en/1562/#organization-id-and-other-information>.
+
+DESCRIPTION
+  Deploy your search application to the Coveo infrastructure.
+Them target `config` if present, must contain the following parameters:
+
+{
```
Comment (ThibodeauJF) [inline packages/cli/core/README.md:1066]: @jpmarceau updated

File: packages/cli/core/README.md [hunk 6]
```diff
@@ -1043,6 +1044,66 @@ EXAMPLES
...
+  "name": "The name of the hosted search page.",
+  "dir": "The directory of the hosted search page.",
+  "htmlEntryFile": {
+    "path": "The path to an HTML file containing the HTML markup of the hosted page."
+  },
+  "javascriptEntryFiles": [
+    {
+      "path": "The path to a bundled Javascript file.",
+      "isModule": "Whether the inline code should be treated as a JavaScript module. If this property is true, the type property will be set to module on the script tag."
```
Comment (jpmarceau) [inline packages/cli/core/README.md:1075]:
```suggestion
      "isModule": "Whether the inline code should be treated as a JavaScript module. If this property is true, the type property will be set to `module` on the script tag."
```

File: packages/cli/core/README.md [hunk 7]
```diff
@@ -1043,6 +1044,66 @@ EXAMPLES
...
+    }
+  ],
+  "javascriptUrls": [
+    {
+      "path": "The URL of the JavaScript source file.",
+      "isModule": "Whether the inline code should be treated as a JavaScript module. If this property is true, the type property will be set to module on the script tag."
```
Comment (jpmarceau) [inline packages/cli/core/README.md:1081]:
```suggestion
      "isModule": "Whether the inline code should be treated as a JavaScript module. If this property is true, the type property will be set to `module` on the script tag."
```

File: packages/cli/core/README.md [hunk 8]
```diff
@@ -1043,6 +1044,66 @@ EXAMPLES
...
+    }
+  ],
+  "cssEntryFiles": [
+    {
+      "path": "The path to a bundled CSS file."
+    }
+  ],
+  "cssUrls": [
+    {
+      "path": "The URL of CSS stylesheet."
```
Comment (jpmarceau) [inline packages/cli/core/README.md:1091]:
```suggestion
      "path": "The URL of the CSS stylesheet."
```

File: packages/cli/core/README.md [hunk 9]
```diff
@@ -1043,6 +1044,66 @@ EXAMPLES
...
+FLAGS
+  -p, --pageId=7944ff4a-9943-4999-a3f6-3e81a7f6fb0a  The hosted page ID.
+  -c, --config=./configs/myconfig.json [default: coveo.deploy.json] The path to the configuration file.
+  -o, --organization=myOrgID  The identifier of the organization inside which to perform operations. If not specified, the CLI logs you in to
+                              the first available organization. See
+                              <https://docs.coveo.com/en/1562/#organization-id-and-other-information>.
+
+DESCRIPTION
+  Deploy your search application to the Coveo infrastructure.
+Them target `config` if present, must contain the following parameters:
```
Comment (jpmarceau) [inline packages/cli/core/README.md:1064]:
```suggestion
The target `config` if present, must contain the following parameters:
```

File: packages/cli/core/README.md (hunk 5 above)
Comment (jpmarceau) [reply packages/cli/core/README.md:1066]: Thank you!

**Pull request merged**